pkg/scheduler/api/shared_device_pool.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	GPUSharingDevice = "GpuShare"
+)
+
+type SharedDevicePool interface {
+	//following two functions used in node_info
+	AddResource(pod *v1.Pod)
+	SubResource(pod *v1.Pod)
+
+	//following four functions used in predicate
+	RequestInPod(pod *v1.Pod) bool
```

Comment:
suggest to change `RequestInPod` to `HasDeviceRequest` or some other name that is easy to understand.